publish/publish.proj

```diff
@@ -7,12 +7,19 @@
   <UsingTask TaskName="ListAzureBlobs" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.CloudTestTasks.dll" />
   <UsingTask TaskName="UploadToAzure" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.CloudTestTasks.dll" />
 
+  <PropertyGroup>
+    <FeedsDir Condition="'$(RunningOnCore)' != 'true'">$(ToolsDir)Microsoft.Dotnet.Build.Tasks.Feed/net46/</FeedsDir>
+    <FeedsDir Condition="'$(RunningOnCore)' == 'true'">$(ToolsDir)microsoft.dotnet.build.tasks.feed/netstandard1.5/</FeedsDir>
```

Comment:
I wish we would start doing this differently.

Instead of following this old pattern that we use in BuildTools, where we do a bunch of restoring, copying, probing, etc manually, we would take advantage of the MSBuild features built into NuGet.  This isn't how we tell customers to provide custom build tasks.

Instead, it would be better if the `microsoft.dotnet.build.tasks.feed` package had a `build` folder that injected Tasks/Targets/etc into projects that `<PackageReference>` to it.

This is the normal way our customers inject custom MSBuild constructs into projects.

/cc @weshaggard

Reply:
See https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#including-msbuild-props-and-targets-in-a-package for more info on how this works.

Basically, if you have .targets and .props files in the `build` folder of your NuGet package, they are automatically included in the project.

Reply:
@karajas is this something you can implement in the buildtools package?

Reply:
The thing is this will break how we've told CLI and CoreCLR how to update. Do we want this change only for core-setup?

Reply:
Do we have any examples of props being imported via a PackageReference in core-setup? I agree it's the right thing to do but we'd really like to unblock this today if possible

Reply:
I would like to restructure the package such that we can get a common layout and support targets for this task library but from init-tools.msbuild project not necessary in the publish.proj. This is not a normal project and so we aren't going to call restore on it in order to use a PackageReference directly in this project.

Reply:
It allows the task dll loading to be handled by a targets file that comes in the package.

Reply:
This has now been updated.